docs/documentation/functions/index.md

```diff
@@ -97,3 +97,28 @@ describe 'ensure_resource' do
   end
 end
 {% endhighlight %}
+
+## Mocking a function for use in all tests
+    
+You may not need to test a function itself, but a calling class may require the function
+to return a value. You can mock the function once and make it available to all test.
+Modify *spec/spec_helper_local.rb* and use `Puppet::Parser::Functions.newfunction()`
+to add the function inside of an `RSpec.configure` block. For example:
+    
+{% highlight ruby %}
+RSpec.configure do | c|
+  c.before :each do
+    # The vault_lookup function takes a single argument and returns a hash with three keys
+    Puppet::Parser::Functions.newfunction(:vault_lookup, type: :rvalue) do |_args|
```

Comment:
There's some discussion of that in slack and how it would apply to tests, there's not a lot of clarity there and the "obvious" code doesn't seem to work. Very happy to move to that if we can determine the correct syntax to use.

Reply:
I'm looking at the function testing code and `Puppet.override` looks promising. It is a private API of Puppet, but looks promising.
https://github.com/puppetlabs/rspec-puppet/blob/f4c5c71dd0271f42891d76a0d017331da7f0c126/lib/rspec-puppet/example/function_example_group.rb#L87-L108

Another problem you may have with the modern API is that it prefers namespaced functions, or there's a difference in priority.